lib/actions/apiV2.js

```diff
@@ -1144,6 +1145,16 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
                               config.transitOperators
                             ),
                             { color: leg?.route?.color, mode: leg.mode }
+                          ).split('#')?.[1],
+                          textColor: getRouteTextColorBasedOnSettings(
```

Comment:
put this in a const at the top of the map. also refactor `getRouteOperator` into its own variable so it isn't repeated.